docs/operators/s3outputmodule.md

```diff
@@ -0,0 +1,199 @@
+
+-   ***secretAccessKey***   -   String
+    -   Mandatory Parameter
+    -   Specifies the AWS secret access key to access Amazon S3 and have permissions to access the specified bucket.
+    
+-   ***endPoint***  -   String
+    -   Endpoint is the url that is entry point for a web service. Specify the valid endpoint to access S3 bucket.
+
+-   ***bucketName***    -   String
+    -   Mandatory Parameter
+    -   S3 buckets are used to store objects which consists of data and metadata that describes the data. Specify the name of the bucket.
+
+-   ***outptuDirectoryPath***   -   String
+    -   Mandatory Parameter
+    -   Specifies the path of the output directory. 
+
+-   ***mergerCount***   -   int
+    -   Specify the number of instances of S3FileMerger operator.
+    -   Default value = 1
+    
+-   ***timeOutWIndowCount***    -   int
+    -   Specify the count as streaming windows. This value will set to the operators in this module because the operators in this module is mostly interacts with the Amazon S3. Due to this reason, window id of these operators might be lag behind with the upstream operators.
+    -   Default value = 6000
+
+### Ports
+---------
+
+-   ***filesMetadataInput***    -   AbstractFileSplitter.FileMetadata
+    -   Input port for files metadata.
+    -   Mandatory
+
+-   ***blocksMetadataInput***   -   BlockMetadata.FileBlockMetadata
+    -   Input port for blocks metadata.
+    -   Mandatory
+    
+-   ***blockData*** -   AbstractBlockReader.ReaderRecord<Slice>
+    -   Input port for blocks data.
+    -   Mandatory
+    
+### Application Example
+-----------------------
+
+Please refer [Example](https://github.com/DataTorrent/examples/tree/master/tutorials/s3output) for S3OutputModule sample application.
+
+### Partitioning
+----------------
+
+Partitioning the module means that operator/operators in the module can able to be partitioned.
+
+#### Stateless Partitioning
+---------------------------
+
+Partitioning the operator in module can be achieved as follows:
+
+-   ***S3InitateFileUploadOperator***
+    -   Partition of this operator is not exposed from module. But, still can be achieved as follows:
+        ```xml
+        <property>
+          <name>dt.operator.{ModuleName}#InitiateUpload.attr.PARTITIONER</name>
+          <value>com.datatorrent.common.partitioner.StatelessPartitioner:{N}</value>
+        </property>     
+        ```
+
+        where {ModuleName} is the name of the S3OutputModule and
+              {N} is the number of static partitions.
+        Above lines will partition S3InitateFileUploadOperator statically {N} times. 
+
+-   ***S3BlockUploadOperator***
+    -   Locality of S3BlockUploadOperator with upstream operator must set to PARTITION_PARALLEL for performance benefits by avoiding serialization/deserialization of objects. So, Partitioning of this operator depends on upstream operator which is of type FSInputModule/BlockReader. 
```

Comment:
Locality is for a stream; looks like this operator has multiple upstream operators ? If so, need to clarify which stream is intended here. Generally this paragraph is difficult to follow and should be explained in greater detail.